### PR TITLE
Make the test more expressive

### PR DESCRIPTION
**Pull request by kenzan100**
My attempt to understand this library better.

* Include readable arguments to express sub-queue intention.

### CODE REVIEW

File: test/test_client.rb [hunk 1]
```diff
@@ -31,15 +31,23 @@ class PrioritizedWorker
...
+      assert PrioritizedWorker.perform_async('user_2', 'enqueued_third')
+
+      assert_equal 3, Sidekiq::PriorityQueue::Queue.new().size
+
+      jobs_by_priority = Sidekiq
+        .redis { |c| c.zrange('priority-queue:default', 0, 2, withscores: true) }
+        .map   { |args, priority| [priority, JSON.parse(args)['args']] }
+
+      assert = [
+        [1.0, ["user_1", "enqueued_first"]],
```
Comment (jamatthews) [inline test/test_client.rb]: This works but it's based on a wrong assumption. The ordering of keys with the same score is lexicographical (sorta alphabetical): https://redis.io/commands/zrange. We should avoid any tests which check the order of jobs with the same scores.

This test was also written before Velina and I built the API to match the Sidekiq API here: https://github.com/chartmogul/sidekiq-priority_queue/blob/master/lib/sidekiq/priority_queue/api.rb#L21

File: test/test_client.rb [hunk 2]
```diff
@@ -31,15 +31,23 @@ class PrioritizedWorker
...
+      assert PrioritizedWorker.perform_async('user_1', 'enqueued_second')
+      assert PrioritizedWorker.perform_async('user_2', 'enqueued_third')
+
+      assert_equal 3, Sidekiq::PriorityQueue::Queue.new().size
+
+      jobs_by_priority = Sidekiq
+        .redis { |c| c.zrange('priority-queue:default', 0, 2, withscores: true) }
+        .map   { |args, priority| [priority, JSON.parse(args)['args']] }
+
+      assert = [
```
Comment (jamatthews) [inline test/test_client.rb]: This works but it's based on a wrong assumption. The ordering of keys with the same score is lexicographical (sorta alphabetical): https://redis.io/commands/zrange. We should avoid any tests which check the order of jobs with the same scores.

This test was also written before Velina and I built the API to match the Sidekiq API here: https://github.com/chartmogul/sidekiq-priority_queue/blob/master/lib/sidekiq/priority_queue/api.rb#L21

Comment (jamatthews) [inline test/test_client.rb]: This works but it's based on a wrong assumption. The ordering of keys with the same score is lexicographical (sorta alphabetical): https://redis.io/commands/zrange. We should avoid any tests which check the order of jobs with the same scores.

This test was also written before Velina and I built the API to match the Sidekiq API here: https://github.com/chartmogul/sidekiq-priority_queue/blob/master/lib/sidekiq/priority_queue/api.rb#L21

File: test/test_client.rb (hunk 1 above)
Comment (kenzan100) [reply test/test_client.rb]: I understand, that we shouldn’t rely on the Sidekiq internal behavior of persisted JSON job such as

```
{"class":"TestClient::PrioritizedWorker","args":["user_2","enqueued_third"],"retry":true,"queue":"default","subqueue":"user_2","jid":"67abe4583b3e25ffb6f00cb0","created_at":1540198161.9649086}
```

to make lexicographical ordering assumption work.

but to me, testing the behavior of `zrange` while having multiple keys under the same score actually is the *core* of this library. So, I would like to test this (having multiple jobs with different `subqueue` having the same score, which your test wasn’t testing).

How can we do it?

**Pull request merged**